### PR TITLE
Fix barerepo

### PR DESCRIPTION
**Pull request by codingtwinky**
fix: #177 

screen shot 2016-03-18 at 6 08 14 pm

### CODE REVIEW

File: clicktests/test.all.js [hunk 1]
```diff
@@ -8,3 +8,4 @@ require('./test.generic');
 require('./test.noheader');
 require('./test.discard');
 require('./test.branches');
+require('./test.bare')
```
Comment (campersau) [inline clicktests/test.all.js]: Nit: missing semicolon at the end.

File: clicktests/test.bare.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+var environment;
+var testRepoPath;
+
+suite.test('Init', function(done) {
+  environment = new Environment(page);
+  environment.init(function(err) {
+    if (err) return done(err);
+    // testRepoPath = environment.path + '/testrepo';
+    testRepoPath = '/tmp/testrepo'
```
Comment (campersau) [inline clicktests/test.bare.js]: Nit: missing semicolon at the end.

File: source/git-api.js [hunk 3]
```diff
@@ -459,12 +459,14 @@ exports.registerApi = function(env) {
...
-        return gitPromise(['rev-parse', '--is-inside-work-tree'], req.query.path)
-          .catch(function(err) {
-            return 'uninited';
-          }).then(function(result) {
-            if (result.toString().indexOf('true') == -1) return 'uninited';
-            else return 'inited';
+        return gitPromise.revParse(req.query.path, '--is-inside-work-tree')
+          .then(function(isWorkingDir) {
+            if (isWorkingDir) {
+              return 'inited'
```
Comment (campersau) [inline source/git-api.js]: Nit: missing semicolon at the end.

File: source/git-api.js [hunk 4]
```diff
@@ -459,12 +459,14 @@ exports.registerApi = function(env) {
...
+            } else {
+              return gitPromise.revParse(req.query.path, '--is-bare-repository')
+                .then(function(isBareDir) { return isBareDir ? 'bare' : 'uninited'; })
```
Comment (campersau) [inline source/git-api.js]: Nit: missing semicolon at the end.

File: source/git-promise.js [hunk 5]
```diff
@@ -378,4 +378,13 @@ git.commit = function(repoPath, amend, message, files) {
   });
 }
 
+git.revParse = function(repoPath, type) {
+  return git(['rev-parse', type], repoPath)
+    .catch(function(err) {
+      return false;
+    }).then(function(result) {
+      return result.toString().indexOf('true') > -1
```
Comment (campersau) [inline source/git-promise.js]: Nit: missing semicolon at the end.

**Comment by campersau** [comment]
If the stash is not expanded and you change to a bare repository then the stash button is visible:
image

If you then click on the stash button to expand it, it disappears.

**Comment by codingtwinky** [comment]
I wonder if you can tell that I'm work on semicolon less languages...  

Thanks @campersau and will be fixing that soon.

**Comment by codingtwinky** [comment]
FYI, git diff in bare repo is not working.  does anybody know a way around this?  maybe @Ajedi32?

simple `git diff` doesn't work at all.

screen shot 2016-03-22 at 5 45 41 pm

**Comment by Ajedi32** [comment]
When you run `git diff` with no arguments it's comparing against the working directory. (It's checking for unstaged, uncommitted changes in the working directory.) So it doesn't really make sense to use that command in a repo that doesn't _have_ a working directory.

You'll have to be more explicit about what revisions you want it to compare.

**Comment by campersau** [comment]
@codingtwinky the problem isn't the actual `git diff` command created by `diffFile` but the `git.status` call in [`diffFile`](https://github.com/FredrikNoren/ungit/blob/master/source/git-promise.js#L252). Which is only used to determine if the file is new.
`git.status` does a lot of things. E.g. also calling `git diff --numstat` for receiving additions / deletions which will fail in bare repositories for the reasons described by @Ajedi32 

The actual diff command used by `diffFile` is already correct and will look like this:
`git diff f4f439e2^ f4f439e2 -- "components/repository/repository.js"`

So there are two solutions which come to my mind:
1. Handle failures of the `git.status` call and proceed with the actual diff logic.
2. Add a parameter to `git.diffFile` and do not call `git.status` at all if the parameter is set.

**Comment by codingtwinky** [comment]
Merging this for now.  created #728 and will be fixing it later

**Pull request merged**